crates/threshold-signature-server/src/user/api.rs

```diff
@@ -325,21 +330,31 @@ async fn setup_dkg(
             .try_into()
             .map_err(|_| UserErr::AddressConversionError("Invalid Length".to_string()))?;
         let sig_request_address = SubxtAccountId32(*address_slice);
-        let user_details =
-            get_registering_user_details(&api, &sig_request_address.clone(), rpc).await?;
+        let mut key_visibility = KeyVisibility::Public;
+        // If master key set visibility to public
+        if sig_request_account != master_key.encode() {
+            let user_details =
+                get_registering_user_details(&api, &sig_request_address.clone(), rpc).await?;
+            key_visibility = user_details.key_visibility.0;
+        };
```

Comment:
```suggestion
        let key_visibility = if sig_request_account == master_key.encode() {
            KeyVisibility::Public
        } else {
            let user_details =
                get_registering_user_details(&api, &sig_request_address.clone(), rpc).await?;
            user_details.key_visibility.0
        };

```

This makes the assignment a bit more clear, and we can also get rid of that comment